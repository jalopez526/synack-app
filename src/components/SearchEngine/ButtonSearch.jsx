import React from "react";
import Button from "../Button";

const ButtonSearch = ({ children, ...rest }) => {
  return (
    <Button data-testid="submit-button" className="button" {...rest}>
      {children}
    </Button>
  );
};

export default ButtonSearch;
