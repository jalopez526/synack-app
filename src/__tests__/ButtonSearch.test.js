import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import ButtonSearch from "../Components/SearchEngine/ButtonSearch";

describe("Testing ButtonSearch component", () => {
  const onClick = jest.fn();

  const Button = <ButtonSearch onClick={onClick}>Search</ButtonSearch>;
  it("renders correctly", () => {
    const tree = renderer.create(Button).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render button title", () => {
    const { getByText } = render(Button);
    expect(getByText(/Search/)).toBeInTheDocument();
  });

  it("should call OnClick function", () => {
    const { getByTestId } = render(Button);
    fireEvent.click(getByTestId("submit-button"));
    expect(onClick).toHaveBeenCalled();
  });
});
