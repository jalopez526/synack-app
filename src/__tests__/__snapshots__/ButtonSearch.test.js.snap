// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ButtonSearch component renders correctly 1`] = `
<button
  className="button"
  data-testid="submit-button"
  onClick={[MockFunction]}
>
  Search
</button>
`;
